ext/date/php_date.c

```diff
@@ -5327,7 +5327,20 @@ static zval *date_period_read_property(zval *object, zval *member, int type, voi
 /* {{{ date_period_write_property */
 static void date_period_write_property(zval *object, zval *member, zval *value, void **cache_slot)
 {
-	zend_throw_error(NULL, "Writing to DatePeriod properties is unsupported");
+	if (!strcmp(Z_STRVAL_P(member), "recurrences")
```

Comment:
These checks should be done on `name` after the `zval_get_string` call -- you can't assume that `member` is a string before that.

I'd also suggest to use `zend_string_equals_literal(name, "recurrences")` etc, to make sure the comparisons are binary-safe.